Source/Scenes/Onboarding/OnboardingInteractor.swift

```diff
@@ -0,0 +1,154 @@
+                Topic(id: UInt.random(in: UInt.min...UInt.max).description, name: "Minim veniam", imageUrl: "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQ9_JWAFA_Q8kQitjXBwRtNUw5xOSKC_SF_rbZOTriNOMncfrM0Ww"),
+                Topic(id: UInt.random(in: UInt.min...UInt.max).description, name: "Quis nostrud exercitation", imageUrl: "https://cdn.cnn.com/cnnnext/dam/assets/190121090951-04-blood-moon-global-01212019-exlarge-169.jpg"),
+                Topic(id: UInt.random(in: UInt.min...UInt.max).description, name: "Ullamco laboris", imageUrl: "https://media.mnn.com/assets/images/2016/11/closeup-baby-robin-beak-open.jpg.653x0_q80_crop-smart.jpg")
+                ].shuffled()))
+
+            self.subtopicsDidFetch()
+        }
+    }
+
+    func topicsDidFetch() {
+        presenter.presentTopicsDidDownload(
+            response: Event.TopicsDidDownload.Response(
+                topics: sceneModel.topics,
+                subtopics: sceneModel.subtopics,
+                selectedTopicsIds: sceneModel.selectedTopicsIds
+            )
+        )
+    }
+
+    func subtopicsDidFetch() {
+        presenter.presentSubtopicsDidDownload(
+            response: Event.SubtopicsDidDownload.Response(
+                topics: sceneModel.topics,
+                subtopics: sceneModel.subtopics,
+                selectedTopicsIds: sceneModel.selectedTopicsIds
+            )
+        )
+    }
+}
+
+// MARK: - Interactor
+extension Onboarding.InteractorImp: Onboarding.Interactor {
+
+    func onViewDidLoad(request: Event.ViewDidLoad.Request) {
+        presenter.presentViewDidLoad(response: Onboarding.Event.ViewDidLoad.Response(numberForSelect: sceneModel.numberForSelect))
+        fetchTopics()
+    }
+
+    func onSelectTopic(request: Event.DidSelectTopic.Request) {
+        if sceneModel.selectedTopicsIds.contains(request.topicId) {
+            if sceneModel.topics.contains(where: { $0.id == request.topicId }) {
+                if let (_, subtopics) = sceneModel.subtopics.first(where: { $0.topicId == request.topicId }) {
+                    sceneModel.selectedTopicsIds.removeAll(where: { topicId in
+                        subtopics.contains { topic in topic.id == topicId }
+                    } )
+
+                    sceneModel.subtopics.removeAll { $0.topicId == request.topicId }
+                }
+            }
+
+            sceneModel.selectedTopicsIds.removeAll { $0 == request.topicId}
+        } else {
+            sceneModel.selectedTopicsIds.append(request.topicId)
+            if sceneModel.topics.contains(where: { $0.id == request.topicId }) {
+                fetchSubtopics(with: request.topicId)
+            }
+        }
+
+        sceneModel.isStartEnabled = sceneModel.selectedTopicsIds.count >= sceneModel.numberForSelect
+
+        presenter.presentSelectTopic(
+            response: Event.DidSelectTopic.Response(
```

Comment:
This can be break to 2 separate responses: first `TopicsDidUpdate` (from the previous review comment), second `ShowGetStartedButton`.